data-prepper-plugins/geoip-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/extension/databasedownload/DatabaseReaderCreateTest.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.extension.databasedownload;
+
+import com.maxmind.geoip2.DatabaseReader;
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Path;
+
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class DatabaseReaderCreateTest {
+    @Mock
+    private Path path;
+
+    @Test
+    void createLoaderTest() throws IOException {
+        final String testFileURL = "https://github.com/maxmind/MaxMind-DB/raw/main/test-data/GeoLite2-City-Test.mmdb";
+        final File file = File.createTempFile( "GeoIP2-City-Test", ".mmdb");
+
+        final BufferedInputStream in = new BufferedInputStream(new URL(testFileURL).openStream());
```

Comment:
I have tried it with mock URL but looks like the DatabaseReader needs an actual mmdb file, it's validating some metadata. So I decided to use actual test mmdb file. Only way to avoid this is not to test the DatabaseReader build.